GabijaBern/assignment1/SingleLinkList.java

```diff
@@ -0,0 +1,76 @@
+//Q2 Implement an algorithm to find the kth to last element of a singly linked list
+public class SingleLinkList<Type> 
+{
+	private Node<Type> head = null;

```

Comment:
Style: Initialising to `null` and `0` is the default, so it is usually omitted.

Reply:
Yes, but my teacher always says not to count on default values and make the code clear :) should I still change it?

Reply:
I think this can be a matter of personal preference, and depend on your team's or company's style guide. The important thing is to always be consistent. If you sometimes use it, and sometimes don't then people don't know what you intended. Personally I agree with your teacher, but I know some teams at Google do the opposite and never add it.